var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1127,3 +1127,22 @@ def remove_files_from_view(self, view, merge_map):
                 view.remove_file(src, dst)
             else:
                 os.remove(dst)
+
+    def test(self):
+        # do not use self.command because we are also testing the run env
+        exe = self.command.name
```

Comment:
```suggestion
        exe = self.spec['python'].command.name
```

Reply:
These appear to be equivalent.  Why is the one you're suggesting better?

Reply:
They are equivalent but misleading. `self.spec['foo'].command` works for any package. `self.command` only works for packages that override the default command handler. Same for libs and headers. I talked to @alalazo about this for a different PR and we decided it would be best to be uniform and always use `self.spec['foo'].whatever`.